### PR TITLE
Functional tests: update dev container build Fix #1156

### LINKED ISSUE #1156: Functional tests: dev container fails on userpass authentication
Motivation
----------
When spooling up a new dev environment using the `rucio/rucio-dev` many tests fail. The first message is an exception that authentication with username and password fails.

Modification
------------
- [x] Update Dockerfile
   - [x] Update packages installed by yum
   - [x] Install all requirements with pip (such that `tools/run_tests_docker.sh` may be used)
   - [x] Remove python-ipaddress package to handle pip dependencies
- [x] Update `rucio.cfg` to streamline with travis container
- [x] Change mysql credentials in `docker-compose.yml` to rucio.cfg

### PR DESCRIPTION
**Pull request by berghaus**
Functional tests: update dev container build Fix #1156

### CODE REVIEW

**Comment by berghaus** [comment]
Something odd is going on here. One of the tests consistently seems to end up with authentication issues during the test, while the other all pass. I had seen these authentication issues in a misconfigured dev container in the past ... I wonder if I can reproduce this.

**Comment by vingar** [comment]
I vaguely remember something like that due to wrong CA certificate(not the same on the server) in the configuration and had to fix it for the travis test.

**Comment by bari12** [comment]
I re-executed the test now and it passed. Maybe this is one of the occasional hickups we have on travis sometime...

**Comment by berghaus** [comment]
Thanks Martin!

File: etc/docker/dev/rucio.cfg [hunk 1]
```diff
@@ -53,9 +37,9 @@ gss_identity = ddmlab@CERN.CH
 gss_email = ph-adp-ddm-lab@cern.ch
 
 [monitor]
-carbon_server = localhost
+carbon_server = rucio-graphite-int.cern.ch
```
Comment (mlassnig) [inline etc/docker/dev/rucio.cfg]: won't this spam our graphite server with random people firing up dev containers?

Comment (berghaus) [reply etc/docker/dev/rucio.cfg]: Indeed. What is the best way to turn this off entirely for the dev container? Remove the config or point to localhost?

Comment (mlassnig) [reply etc/docker/dev/rucio.cfg]: leaving it at localhost is fine. maybe later we can even include graphite in the dev container (so people can actually look at the metrics). i think @tbeerman already did that.

Comment (tbeerman) [reply etc/docker/dev/rucio.cfg]: yes, I've played around with it. You can just add the `graphiteapp/graphite-statsd` image in docker-compose and link it. That should do it.

Comment (berghaus) [reply etc/docker/dev/rucio.cfg]: Thanks @tbeerman I'll do that :-)

Comment (bari12) [reply etc/docker/dev/rucio.cfg]: Ping 😄

**Review by vingar — APPROVED** [review]

**Pull request merged**